network-protection/network-protection-impl/src/main/java/com/duckduckgo/networkprotection/impl/waitlist/NetPInviteCodeActivity.kt

```diff
@@ -0,0 +1,161 @@
+ * limitations under the License.
+ */
+
+package com.duckduckgo.networkprotection.impl.waitlist
+
+import android.os.Bundle
+import androidx.activity.result.ActivityResult
+import androidx.activity.result.contract.ActivityResultContracts
+import androidx.lifecycle.Lifecycle
+import androidx.lifecycle.flowWithLifecycle
+import androidx.lifecycle.lifecycleScope
+import com.duckduckgo.anvil.annotations.ContributeToActivityStarter
+import com.duckduckgo.anvil.annotations.InjectWith
+import com.duckduckgo.app.global.DuckDuckGoActivity
+import com.duckduckgo.app.statistics.pixels.Pixel
+import com.duckduckgo.di.scopes.ActivityScope
+import com.duckduckgo.mobile.android.ui.view.gone
+import com.duckduckgo.mobile.android.ui.view.show
+import com.duckduckgo.mobile.android.ui.viewbinding.viewBinding
+import com.duckduckgo.navigation.api.GlobalActivityStarter
+import com.duckduckgo.navigation.api.GlobalActivityStarter.ActivityParams
+import com.duckduckgo.navigation.api.getActivityParams
+import com.duckduckgo.networkprotection.api.NetPInviteCodeScreenNoParams
+import com.duckduckgo.networkprotection.api.NetworkProtectionManagementScreenNoParams
+import com.duckduckgo.networkprotection.api.NetworkProtectionWaitlist.NetPWaitlistState.InBeta
+import com.duckduckgo.networkprotection.api.NetworkProtectionWaitlist.NetPWaitlistState.JoinedWaitlist
+import com.duckduckgo.networkprotection.api.NetworkProtectionWaitlist.NetPWaitlistState.NotUnlocked
+import com.duckduckgo.networkprotection.api.NetworkProtectionWaitlist.NetPWaitlistState.PendingInviteCode
+import com.duckduckgo.networkprotection.impl.R
+import com.duckduckgo.networkprotection.impl.about.NetPTermsScreenNoParams
+import com.duckduckgo.networkprotection.impl.databinding.ActivityNetpInviteCodeBinding
+import com.duckduckgo.networkprotection.impl.waitlist.NetPInviteCodeActivity.Companion.NetPInviteCodeScreenWithOriginPixels
+import com.duckduckgo.networkprotection.impl.waitlist.NetPInviteCodeViewModel.Command
+import com.duckduckgo.networkprotection.impl.waitlist.NetPInviteCodeViewModel.Command.EnterInviteCode
+import com.duckduckgo.networkprotection.impl.waitlist.NetPInviteCodeViewModel.Command.OpenNetP
+import com.duckduckgo.networkprotection.impl.waitlist.NetPInviteCodeViewModel.Command.OpenTermsScreen
+import com.duckduckgo.networkprotection.impl.waitlist.NetPInviteCodeViewModel.ViewState
+import com.duckduckgo.networkprotection.impl.waitlist.NetPWaitlistRedeemCodeActivity.Launch.NetPWaitlistRedeemCodeScreenNoParams
+import com.duckduckgo.networkprotection.impl.waitlist.store.NetPWaitlistRepository
+import javax.inject.Inject
+import kotlinx.coroutines.flow.launchIn
+import kotlinx.coroutines.flow.onEach
+
+@InjectWith(ActivityScope::class)
+@ContributeToActivityStarter(NetPInviteCodeScreenNoParams::class)
+@ContributeToActivityStarter(NetPInviteCodeScreenWithOriginPixels::class)
+class NetPInviteCodeActivity : DuckDuckGoActivity() {
```

Comment:
NIT: Name of this class is a bit confusing as I confused this with `NetPWaitlistRedeemCodeActivity`. I think this is better called along the lines of `NetPInvitedActivity` or `NetPGetStartedActivity`.

Reply:
Will rename yup